cluster/ubuntu/util.sh

```diff
@@ -346,99 +347,229 @@ function kube-up() {
+    saltbase/salt/generate-cert/make-ca-cert.sh \
+    ubuntu/${KUBE_CONFIG_FILE:-config-default.sh} \
+    ubuntu/util.sh \
+    ubuntu/minion/* \
+    ubuntu/master/* \
+    ubuntu/reconfDocker.sh \
+    ubuntu/binaries/master/ \
+    ubuntu/binaries/minion \
+    "${MASTER}:~/kube"
+  
+  EXTRA_SANS=(
+    IP:${MASTER_IP}
+    IP:${SERVICE_CLUSTER_IP_RANGE%.*}.1
+    DNS:kubernetes
+    DNS:kubernetes.default
+    DNS:kubernetes.default.svc
+    DNS:kubernetes.default.svc.cluster.local
+  )
+  
+  EXTRA_SANS=$(echo "${EXTRA_SANS[@]}" | tr ' ' ,)
+
+  # remote login to the master/node and configue k8s
+  ssh $SSH_OPTS -t "$MASTER" "
+    source ~/kube/util.sh
+     
+    setClusterInfo
+    create-etcd-opts
+    create-kube-apiserver-opts \
+      '${SERVICE_CLUSTER_IP_RANGE}' \
+      '${ADMISSION_CONTROL}' \
+      '${SERVICE_NODE_PORT_RANGE}'
+    create-kube-controller-manager-opts '${NODE_IPS}'
+    create-kube-scheduler-opts
+    create-kubelet-opts \
+      '${MASTER_IP}' \
+      '${MASTER_IP}' \
+      '${DNS_SERVER_IP}' \
+      '${DNS_DOMAIN}'
+    create-kube-proxy-opts '${MASTER_IP}'
+    create-flanneld-opts '127.0.0.1'
+    
+    sudo -E -p '[sudo] password to start master: ' -- /bin/bash -c ' 
+      cp ~/kube/default/* /etc/default/ 
+      cp ~/kube/init_conf/* /etc/init/ 
+      cp ~/kube/init_scripts/* /etc/init.d/
+      
+      groupadd -f -r kube-cert
+      \"${PROXY_SETTING}\" ~/kube/make-ca-cert.sh \"${MASTER_IP}\" \"${EXTRA_SANS}\"
+      mkdir -p /opt/bin/ 
+      cp ~/kube/master/* /opt/bin/
+      cp ~/kube/minion/* /opt/bin/
+
+      service etcd start
+      FLANNEL_NET=\"${FLANNEL_NET}\" ~/kube/reconfDocker.sh ai
+      '" || {
+      echo "Deploying master and node on machine ${MASTER_IP} failed"
+      exit 1
+  }    
 }
 
 # Delete a kubernetes cluster
-function kube-down {
+function kube-down() {
+  
   export KUBECTL_PATH="${KUBE_ROOT}/cluster/ubuntu/binaries/kubectl"
-  source "${KUBE_ROOT}/cluster/ubuntu/${KUBE_CONFIG_FILE-"config-default.sh"}"
+  source "${KUBE_ROOT}/cluster/ubuntu/${KUBE_CONFIG_FILE:-config-default.sh}"
   source "${KUBE_ROOT}/cluster/common.sh"
+
   tear_down_alive_resources
 
   local ii=0
   for i in ${nodes}; do
-    {
-      echo "Cleaning on node ${i#*@}"
       if [[ "${roles[${ii}]}" == "ai" || "${roles[${ii}]}" == "a" ]]; then
-        ssh $SSH_OPTS -t $i 'pgrep etcd && sudo -p "[sudo] password to stop master: " service etcd stop && sudo rm -rf /infra*; 
-          sudo rm -rf /opt/bin/etcd* /etc/init/etcd.conf /etc/init.d/etcd /etc/default/etcd'
+        echo "Cleaning on master ${i#*@}"
+        ssh $SSH_OPTS -t "$i" "
+          pgrep etcd && \
+          sudo -p '[sudo] password to stop master: ' -- /bin/bash -c '
+            service etcd stop
+
+            rm -rf \
+              /opt/bin/etcd* \
+              /etc/init/etcd.conf \
+              /etc/init.d/etcd \
+              /etc/default/etcd
+      
+            rm -rf /infra*
+            rm -rf /srv/kubernetes
```

Comment:
I believe kube-down should do all of the clean-up work. What's involved this directory is legacy files created by `make-ca-cert.sh`